### DISCUSSION TITLE
Issue with rviz (for assignment 1)

### DISCUSSION THREAD

**Discussion by chaidop** [gstavrinos/ai_msc_robotic_systems_assignments#29] [Q&A]
Hello! Προσπαθω να ανοιξω το rviz για να δω στον αρχικο κωδικα το simulation, αλλα δεν εμφανιζεται κατι στο display του. Έχω κάνει check το TF και το Image, αλλα δεν εμφανιζει κατι... Ισως φταιει η δικη μου παραβλεψη σε καποιο comand, παντως τα βηματα που ακολουθησα για να τρεξω το Node και να δω τη προσομοιωση ηταν να κανω colcon build μεσα στον φακελο του assignments, να κάνω source το /opt/ros/humble/setup.bash και το ros_ws/install/setup.bash, να τρεξω το launch file "ros2 launch tt_ball_locator ball_locator.launch.py ", να τρεξω παλι το node με "ros2 run tt_ball_locator tt_ball_locator.py" και στη συνεχεια να ανοιξω τη προσομοιωση με "rviz2". Παρολαυτα, και ενω εχω τσεκαρει το image field, δεν μου εμφανιζεται κατι στην οθονη. Παρελειψα καποιο command? Θενξ!

**Comment by gstavrinos** [comment]
Δες καλύτερα το ερώτημα 3 της πρώτης άσκησης, που σου περιγράφει τί πρέπει να κάνεις για να δοκιμάσεις τον κώδικά σου. Δες τις πληροφορίες που σου δίνονται και το πώς εκτελούμε nodes ή/και launch files, και νομίζω θα είναι ξεκάθαρο. Αν και πάλι δεν βγάλεις άκρη, η επόμενη βοήθεια θα περιλαμβάνει την εντολή που πρέπει να τρέξεις.

**Comment by Xindictus** [comment]
Μια επιπλέον ερώτηση πάνω στο `rviz2`, με κάποιο τρόπο είχαμε ενεργοποιήσει ένα 3d view του environment. Ενεργοποίησα το `PointCloud2` στο `rviz2`, αλλά δεν άλλαξε κάτι. Χρειάζεται κάτι ακόμα;

**Reply by gstavrinos** [reply]
Σιγουρέψου ότι τρέχεις τη σωστή εντολή που ζητείται στο τρίτο ερώτημα της εργασίας. Αν και πάλι δεν βλέπεις κάτι στο rviz, δώσε μας τα βήματα που εκτελείς.

**Reply by Xindictus** [reply]
Η εντολή θεωρώ είναι σωστή γιατί στο 2d plane τα βλέπω όλα, ακόμα και το `tf2` tree. Πριν που ξαναπροσπάθησα τα ίδια βήματα μου έβγαλε μόνο το τραπέζι του ping-pong. Υποπτεύομαι ότι έχει να κάνει με το γεγονός ότι κάνω αλλαγές στο `Assignment1Node` και κάνω restart το simulation με αποτέλεσμα μετά να βλέπω κάποια errors ενδιάμεσα στα logs. Υπάρχει τρόπος να αφήνουμε το simulation να τρέχει και να κάνουμε μόνο το `node` μας restart; Δλδ το sim θα κάνει retry το `on_configure` και `activate` του node ξανά;

**Reply by gstavrinos** [reply]
Οκ, σωστή ακούγεται αφού παίρνεις τα υπόλοιπα από τα δεδομένα. Σχετικά με το τί φαίνεται στο 3D, σκέψου ότι ό,τι βλέπουμε στο rviz είναι όσα ξέρει/βλέπει το ρομπότ, οπότε για να δεις άλλα κομμάτια του χώρου θα πρέπει να κινηθεί το ρομπότ. Έτσι όπως το περιγράφεις φαίνεται να μην έχεις κάποιο πρόβλημα.

Σχετικά με το ευκολότερο restart, δεν έχω υλοποιήσει κάτι τέτοιο. Είναι καλή παρατήρηση, αλλά προτίμησα να τα ενσωματώσω όλα σε μία εντολή από το να σπάσει σε δύο κομμάτια. Θα ανοίξω ένα poll, και αν και οι υπόλοιποι συμφωνούν, θα το σπάσω σε δύο εντολές για ευκολότερο restart.

**Reply by Xindictus** [reply]
Θεωρητικά, δε θα μπορούσαμε να τρέξουμε απλώς με τη σειρά τα παρακάτω;
```bash
ros2 lifecycle set <node> configure
ros2 lifecycle set <node> activate
```

Αυτό βέβαια δε θα τσεκάρει όσα χρειάζονται όπως ότι δε κάνουμε subscribe πριν γίνουμε `active`, αλλά από την άλλη μας το έχεις ήδη έτοιμο αυτό το κομμάτι.

**Reply by gstavrinos** [reply]
Όχι, δεν είναι τόσο απλό, θα το συζητήσουμε όμως.

**Reply by chaidop** [reply]
Καλησπέρα σχετικά με το pointcloud που αναφερει ο Κωνσταντινος, εμενα ε μου δειχνει καν το τραπεζι. Βασικα τιποτα μονο το tf. Τρεχω απλα το launch tt_umpire και launch tt_ball_locator. Πρεπει να ριξω καποια ακομη εντολη;

**Comment by gstavrinos** [comment]
> Μια επιπλέον ερώτηση πάνω στο `rviz2`, με κάποιο τρόπο είχαμε ενεργοποιήσει ένα 3d view του environment. Ενεργοποίησα το `PointCloud2` στο `rviz2`, αλλά δεν άλλαξε κάτι. Χρειάζεται κάτι ακόμα;

Τί ζητάει το τρίτο ερώτημα της εκφώνησης;
